### PR TITLE
Update Vagrant project dependencies and folder structure

### LINKED ISSUE #3: Update Vagrant project to install up-to-date dependencies
Upgrade:

- Terraform
- AWS CLI

Also, ensure that the AWS environment is setup properly for the tooling above to work and update the directory structure.

### PR DESCRIPTION
**Pull request by tnation14**
# OVERVIEW
This PR updates `terraform` and `awscli` versions, and reorganizes Terraform and Ansible resources to conform to the organizational standards set in other repos. The `Vagrantfile` and `scripts/infra` have been updated to reflect this change.

- Bump Terraform version to 0.9.3
- Move `terraform` and `ansible` directories to a `deployment/` folder
- Add Ansible role to install `awscli==11.82.*` and configure the AWS environment.
- Mount `~/.aws` into the VM.

# TESTING
- Follow the `Environment Setup` steps in the `README` to provision the VM.
- `vagrant ssh`, then run `scripits/infra plan` according to the instructions in the `README`. Ensure that a) the command runs successfully and b) no changes are necessary.

Fixes #3

### CODE REVIEW

File: deployment/ansible/roles/gtsite.awscli/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+- name: Install awscli
+  pip: name=awscli version={{ awscli_version }}
```
Comment (hectcastro) [inline deployment/ansible/roles/gtsite.awscli/tasks/main.yml]: We have a role that can be used for installing the AWS CLI: https://github.com/azavea/ansible-aws-cli

File: README.md [hunk 2]
```diff
@@ -19,6 +19,18 @@ available at `transit/`
...
 
+### Environment Setup
+Create an AWS profile called `geotrellis-site`
+```bash
+aws --profile geotrellis-site configure
+```
+
+Create and provision the VM
+```bash
+$ vagrant up --provision
```
Comment (hectcastro) [inline README.md]: I think that it would be good to replace this step with the usual `setup` script that does mostly this.

**Review by hectcastro — CHANGES_REQUESTED** [review]

File: deployment/ansible/roles/gtsite.awscli/tasks/main.yml (hunk 1 above)
Comment (tnation14) [reply deployment/ansible/roles/gtsite.awscli/tasks/main.yml]: Thanks. I knew we had this role but I couldn't find it; I was searching for `ansible-awscli`.

**Comment by tnation14** [comment]
Updated.

File: Vagrantfile [hunk 3]
```diff
@@ -17,13 +17,30 @@ Vagrant.configure("2") do |config|
...
-        ansible.galaxy_roles_path = "/home/vagrant/ansible_galaxy"
+        ansible.playbook = "deployment/ansible/playbook.yml"
+        ansible.galaxy_role_file = "deployment/ansible/roles.yml"
+        ansible.galaxy_roles_path = "deployment/ansible/roles"
+        ansible.groups = {
+            "all" => ["default"],
+            "all:vars" => {"terraform_version"                    => "0.9.3",
+                           "awscli_version"                       => "1.11.*",
+                           "aws_profile"                          => "geotrellis-site",
+                           "java_flavor"                          => "oracle",
```
Comment (hectcastro) [inline Vagrantfile]: Why is Java being installed inside the VM? The goal of this repository is only support AWS infrastructure management (the core column on the whiteboard from yesterday).

**Comment by tnation14** [comment]
Java references have been removed, and `scripts/setup` now only runs `vagrant up --provision`.

**Review by hectcastro — APPROVED** [review]

**Pull request merged**